reconcile/utils/terraform_client.py

```diff
@@ -785,6 +785,56 @@ def validate_db_upgrade(
                     f"{resource_name} to a new major version."
                 )
 
+            blue_green_update = after.get("blue_green_update", {}).get("enabled", False)
+            if blue_green_update:
+                self.validate_blue_green_update_requirements(
+                    account_name,
+                    engine,
+                    after_version,
+                    before["db_parameter_group_name"],
+                    before.get("replica_source"),
+                    region_name,
+                )
+
+    def validate_blue_green_update_requirements(
+        self,
+        account_name,
+        engine,
+        version,
+        parameter_group,
+        replica_source,
+        region_name,
+    ):
+        supported_versions = {
+            "mysql": ["5.7", "8.0.15"],
+            "postgres": ["11.21", "12.16", "13.12", "14.9", "15.4", "16.1"],
```

Comment:
Can we avoid hard coding these lists of versions somehow?

Reply:
I tried to find an api that provided information specific to blue/green deployment compatibility kind of like how we used DescribeDBEngineVersions for a valid upgrade target but I did not find anything. Do you have any other ideas for how we could avoid hard coding the versions?

Reply:
I think the only way here is hard coding.